### ISSUE TITLE
json information

### ISSUE THREAD

**Issue by tt112997** [Coldilocks/ggnn-token-recommendation-pro#1]
Hello, may I ask if you can see what format your data is processed into? What information is in the JSON file. thank you

**Comment by Coldilocks** [comment]
Hi, I've added some examples of the input data under the **data** package, so you can check it out. I hope this will help.

**Comment by tt112997** [comment]
Thank you for your reply. I am looking at the code of code_rec_token_level. It seems that there is no input of this part under the data file you provided. According to your code, what does d["label "] mean? Is it the type of each node? d [" token "] is all token node ids in a code file? Looking forward to your reply!

**Comment by Coldilocks** [comment]
You are right. "label" means the node type, "token" is the node id, and "target" is the node id to be predicted.

**Comment by tt112997** [comment]
in the code "for token in d["token"]"
one “token” is a list ,it means Where is the current node used?
and it's a token_level_rec, does every node need to be predicted?so the targets is all node id or what?
I'm sorry for my repeated questions.

**Comment by Coldilocks** [comment]
Each node has a type id and at least one token, and the list is used to store the token sequence inside a node.

As to your second question, the model only predicts the next token in the last node, so there is only one token id in "target", which is the same as the basic GGNN model.

**Comment by tt112997** [comment]
i've processed cpp file into ast and added some edges into a graph,a node example is (IDENTIFIER,n,i<n),i <n is the ownership statement of n,like you said, the "token" of this node is [j.typeid,<.typeid,n.typeid]?

**Comment by tt112997** [comment]
i've processed cpp file into ast and added some edges into a graph,a node example is (IDENTIFIER,n,i<n),i <n is the ownership statement of n,like you said, the "token" of this node is [j.typeid,<.typeid,n.typeid]?

**Comment by Coldilocks** [comment]
Sorry, I don't quite understand your example. Are you saying that the current node is "i < n"?

**Comment by Coldilocks** [comment]
If so, "i", "<" and "n" are the tokens of the node "i < n", and the "token" list of this node is ["i".token_id, "<".token_id, "n".token_id]. Only the node has the type id, and the token has the token id. Hope this helps you understand.

**Comment by tt112997** [comment]
Is a node a statement? So one node has multiple tokens? Is the graph generated from the AST?I would also like to know that in the code, the embedding of Type and Token is initialized with id. Is it not used to learn the vocab [id] in the future?

**Comment by tt112997** [comment]
Excuse me, are you Chinese? If so, we can communicate in Chinese, so that I can better express my meaning.

**Comment by jianghc2116** [comment]
Excuse me, you said label was the node type id,token is node id, is it node_value_id? Also node_value_id in graph information [node1,edge_type,node2]? node_value_id refers to the ID value in the dictionary.
